src/Og.php

```diff
@@ -192,6 +199,53 @@ public static function isGroupAudienceField(FieldDefinitionInterface $field_conf
   }
 
   /**
+   * Return all the group audience fields in an entity.
+   *
+   * @param string $entity_type_id
+   *   The entity type.
+   * @param string  $bundle
+   *   The bundle name to be checked.
+   * @param string $group_type_id
+   *   Filter list to only include fields referencing a specific group type.
+   * @param string $group_bundle
+   *   Filter list to only include fields referencing a specific group bundle.
+   *   Fields that do not specify any bundle restrictions at all are also
+   *   included.
+   *
+   * @return \Drupal\Core\Field\FieldDefinitionInterface[]
+   *   An array of field definitions, keyed by field name; Or an empty array if
+   *   none found.
+   */
+  public static function getAllGroupAudienceFields($entity_type_id = 'user', $bundle = 'user', $group_type_id = NULL, $group_bundle = NULL) {
+    $return = [];
```

Comment:
And static caching should come after actual caching IMO. This will be a relatively expensive operation to be running all the time. We will want to be a caching some map of dara to retrieve this instead.

This is just an implementation detail though. So can should be a follow up for sure.